Mage.Sets/src/mage/cards/f/FelixFiveBoots.java

```diff
@@ -0,0 +1,100 @@
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.keyword.MenaceAbility;
+import mage.abilities.keyword.WardAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.DamagedEvent;
+import mage.game.events.GameEvent;
+import mage.game.events.NumberOfTriggersEvent;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+/**
+ * @author PurpleCrowbar
+ */
+public final class FelixFiveBoots extends CardImpl {
+
+    public FelixFiveBoots(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{B}{G}{U}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.OOZE, SubType.ROGUE);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(4);
+
+        // Menace
+        this.addAbility(new MenaceAbility(false));
+
+        // Ward {2}
+        this.addAbility(new WardAbility(new GenericManaCost(2), false));
+
+        // If a creature you control dealing combat damage to a player causes a triggered ability of a permanent you control to trigger, that ability triggers an additional time.
+        this.addAbility(new SimpleStaticAbility(new FelixFiveBootsEffect()));
+    }
+
+    private FelixFiveBoots(final FelixFiveBoots card) {
+        super(card);
+    }
+
+    @Override
+    public FelixFiveBoots copy() {
+        return new FelixFiveBoots(this);
+    }
+}
+
+class FelixFiveBootsEffect extends ReplacementEffectImpl {
+
+    FelixFiveBootsEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Benefit);
+        staticText = "If a creature you control dealing combat damage to a player causes a triggered ability " +
+                "of a permanent you control to trigger, that ability triggers an additional time";
+    }
+
+    private FelixFiveBootsEffect(final FelixFiveBootsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public FelixFiveBootsEffect copy() {
+        return new FelixFiveBootsEffect(this);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.NUMBER_OF_TRIGGERS;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        NumberOfTriggersEvent numberOfTriggersEvent = (NumberOfTriggersEvent) event;
+        Permanent sourcePermanent = game.getPermanent(numberOfTriggersEvent.getSourceId());
+        if (sourcePermanent == null || !sourcePermanent.isControlledBy(source.getControllerId())) {
+            return false;
+        }
+
+        GameEvent sourceEvent = numberOfTriggersEvent.getSourceEvent();
+        if (sourceEvent == null
+                || sourceEvent.getType() != GameEvent.EventType.DAMAGED_PLAYER
```

Comment:
That’s wrong. There are other event types like batches.

See WaytaTrainerProdigy with same ability:

![IMG_0148](https://github.com/magefree/mage/assets/8344157/385253b5-a714-41ab-bb07-aef4e0770382)

Reply:
Must use all player related events:

![IMG_0149](https://github.com/magefree/mage/assets/8344157/7e88ee40-9533-43f4-b7a8-536f4c86392a)

Reply:
@JayDi85 Are the changes made in 9d32bb01af3b7fcbd64865839d6546e043b9b23a what you had in mind?

Reply:
Yes, but it must process batches as BatchEvent. That's code will fail:

![shot_240407_040807](https://github.com/magefree/mage/assets/8344157/5af757ea-b36e-415c-b7fc-ee8db9f7775f)

Reply:
Alright, my most recent commit doesn't fix the issue of the isCombatDamage check, because I'm not sure how to go about doing that. I do think it handles batch events in the sense that it goes through each creature involved in the batch until it reaches a valid one (tests pass and works fine in-game, so presumably fine?). I don't fully understand the necessity of using batches, as this card, unlike Umezawa's Jitte, has a pretty cut and dry condition ("deal combat damage to a player", as opposed to "deal combat damage", which raises multiple events). As a result, I can't think of how to write a test to ensure that this is actually working as intended, hence my lack of understanding as to its necessity. I also don't, as mentioned, know how to check that the damage dealt is combat damage.